modules/ingester/flush.go

```diff
@@ -52,28 +62,55 @@ func (i *Ingester) Flush() {
 	}
 }
 
-// FlushHandler calls sweepUsers(true) which will force push all traces into the WAL and force
+// ShutdownHandler handles a graceful shutdown for an ingester. It does the following things in order
+// * Stop incoming writes by exiting from the ring
+// * Flush all blocks to backend
+func (i *Ingester) ShutdownHandler(w http.ResponseWriter, _ *http.Request) {
+	go func() {
+		// lifecycler should exit the ring on shutdown
```

Comment:
Adding some log messages here to show shutdown started and completed would be helpful.